### PR TITLE
Update GoogleTest to resolve issues with newer Visual Studio

### PR DESCRIPTION
**Pull request by davidbeckingsale**
This update maintains our existing fixes to GoogleTest, and removes any that are now unecessary.

### CODE REVIEW

**Comment by kennyweiss** [comment]
@davidbeckingsale -- I will test this against axom.

Could you please also update the patch file in `` blt/thirdparty_builtin/patches`` ?

See the following commit message for how the last one was created:
  https://github.com/LLNL/blt/commit/493676992482c0a7eae87620dede8fd66f836eb8

**Comment by davidbeckingsale** [comment]
Sure thing. I missed the patch file - will do that.

**Comment by kennyweiss** [comment]
Thanks!  I suppose we should remove the old patch file too, since it is no longer relevant.

File: thirdparty_builtin/CMakeLists.txt [hunk 1]
```diff
@@ -100,10 +100,13 @@ if(ENABLE_TESTS)
         blt_append_custom_compiler_flag( FLAGS_VAR gtest_extra_flags
                                          DEFAULT " " 
                                          MSVC  "/wd4251 /wd4275" )
+
+        set (INSTALL_GMOCK Off)
```
Comment (kennyweiss) [inline thirdparty_builtin/CMakeLists.txt]: (minor) We've been using capital letters for ``ON`` and ``OFF`` throughout blt.

Comment (davidbeckingsale) [reply thirdparty_builtin/CMakeLists.txt]: I'll fix this.

**Comment by davidbeckingsale** [comment]
The commit is ea31cb - https://github.com/google/googletest/commit/ea31cb15f0c2ab9f5f5b18e82311eb522989d747

I'll add that to the merge message, should I wait for another approval?

**Comment by kennyweiss** [comment]
Thanks.  @white238 wanted to give it a quick look.  

He mentioned that he disabled something in the previous gtest by commenting it out.  
You might have already taken care of it with the (new?) options  ``INSTALL_GMOCK`` and ``INSTALL_GTEST``

**Comment by davidbeckingsale** [comment]
Yup, @kennyweiss - the `INSTALL_*` options take care of it 😄

**Pull request merged**